examples/raspberry-pi-threads.rs

```diff
@@ -0,0 +1,58 @@
+use navigator_rs::{Navigator, UserLed};
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread::{self, sleep};
+use std::time::Duration;
+
+fn main() {
+    println!("Creating your SensorData cache!");
+    let sensor_data_lock = Arc::new(RwLock::new(navigator_rs::SensorData::default()));
+
+    println!("Creating your navigator module!");
+    let nav = Arc::new(Mutex::new(Navigator::new()));
+
+    println!("Setting up your navigator, ahoy!");
+    nav.lock().unwrap().init();
+
+    // This code block create a thread that updates the `sensor_data_lock` periodically.
+    let sensor_update_thread_lock = sensor_data_lock.clone();
+    let navigator_lock = nav.clone();
+    let _ = thread::spawn(move || {
+        loop {
+            if let Ok(mut write_guard) = sensor_update_thread_lock.write() {
+                *write_guard = navigator_lock.lock().unwrap().read_all();
+                // println!("Updated value: {:?}", *write_guard);
+            }
+            sleep(Duration::from_millis(10000));
+        }
+    });
```

Comment:
Because here we want to differentiate (also to not shadow the original), we add some suffix, usually `_cloned`:
```suggestion
    // This code block creates a thread that updates the `sensor_data_lock` periodically.
    let sensor_data_cloned = sensor_data_lock.clone();
    let nav_cloned = nav.clone();
    let _ = thread::spawn(move || {
        loop {
            if let Ok(mut sensor_data) = sensor_data_cloned.write() {
                *sensor_data = nav_cloned.lock().unwrap().read_all();
                // println!("Updated value: {sensor_data:?}");
            }
            sleep(Duration::from_millis(10000));
        }
    });
```